rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,127 @@
+---
+feature: meson_builds_nix
+start-date: 2022-08-25
+author: Anderson Torres
+co-authors: @p01arst0rm
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Use meson as an alternative build system for the reference implementation of Nix.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the reference implementation of Nix evaluator, as well as its companion toolset generated from the Nix source tree, is built using the typical `./configure` shell script that relies on autoconf, along with standard GNU Make utility.
+
+Over time, this build system has been modified to keep up with the development needs of Nix projec. However, it has reached a state where the build system became clunky and plastered, hard to understand and modify, consequently making improvements to the project as a whole very difficult.
+
+In particular, many changes have been introduced that impact compatibility outside Linux and NixOS niches. These issues can hinder development on other platforms, including but not limited to Unix-like systems.
```

Comment:
Can you provide examples to changes that affect compatibility? Are there issues you can link?

Reply:
@asymmetric bunch of stuff; some of the issues were to do with hardcoded library dependencies that are set up differently on *BSD, the wrong versions of some headers were included on BSD (sometimes headers have names that differ). there were also issues with locating external components iirc.

Reply:
thats a fair point, but more detailed info of these issues would require someone to attempt building nix on multiple different platforms, and take notes. not saying its a bad idea/shouldnt be done but it would take some time

Reply:
https://github.com/NixOS/nix/pull/6991 << example :)

Reply:
Yes, and that PR has a very illuminating comment from me:

> The first commit is just #5145 and again #6015
> 
> 
> Previous efforts that actually used a different tactic include:
> #5185
> #6005
> 
> Issues include:
> #5140
> #5184
> #5781

This issue has been separately discovered by a variety of people building nix for use as a packager manager but not as an OS. Meson would solve the problem for free, as part of providing functionality that Make doesn't have and is apparently needed.